packages/teleport/src/lib/tdp/codec.ts

```diff
@@ -305,16 +306,28 @@ export default class Codec {
   // Throws an error on an invalid or unexpected MessageType value.
   decodeMessageType(buffer: ArrayBuffer): MessageType {
     const messageType = new DataView(buffer).getUint8(0);
-    if (messageType === MessageType.PNG_FRAME) {
-      return MessageType.PNG_FRAME;
-    } else if (messageType === MessageType.CLIPBOARD_DATA) {
-      return MessageType.CLIPBOARD_DATA;
+    if (
+      messageType === MessageType.PNG_FRAME ||
+      messageType === MessageType.CLIPBOARD_DATA ||
+      messageType === MessageType.ERROR
+    ) {
+      return messageType;
     } else {
       // We don't expect to need to decode any other value on the client side
       throw new Error(`invalid message type: ${messageType}`);
     }
   }
 
+  // decodeError decodes a raw tdp ERROR message and returns it as a javascript Error object
+  // | message type (9) | message_length uint32 | message []byte
+  decodeErrorMessage(buffer: ArrayBuffer): Error {
```

Comment:
Better to return a string, not an actual JS Error object. `decodeErrorMessage(buffer: ArrayBuffer): string`